src/main/java/org/spongepowered/api/text/sink/MessageSink.java

```diff
@@ -24,34 +24,42 @@
  */
 package org.spongepowered.api.text.sink;
 
-import org.spongepowered.api.text.Text;
+import static com.google.common.base.Preconditions.checkNotNull;
+
 import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.text.Text;
+
+import java.util.Optional;
 
 /**
  * Represents a function that takes a message and transforms it for distribution to the given targets.
  */
 public abstract class MessageSink {
```

Comment:
This class could be turned into a functional interface since there are neither fields nor final methods anymore.
````java
event.setMessageSink(server::getOnlinePlayers);
````